packages/terra-form/src/NumberField.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Field from './Field';
+import Input from './Input';
+
+const propTypes = {
+  /**
+   * The defaultValue of the input element
+   */
+  defaultValue: PropTypes.number,
+  /**
+   * Error message for when the input is invalid
+   */
+  error: PropTypes.node,
+  /**
+   *  Help element to display with the input
+   */
+  help: PropTypes.node,
+  /**
+   * Additional attributes to attach to the number input
+   */
+  inputAttrs: PropTypes.object,
+  /**
+   * Whether the field is inline
+   */
+  isInline: PropTypes.bool,
+  /**
+   * Label of the input
+   */
+  label: PropTypes.node,
+  /**
+   * Maximum value allowed for the input
+   */
+  max: PropTypes.number,
+  /**
+   * Minimum value allowed for the input
+   */
+  min: PropTypes.number,
+  /**
+   * Name of the input attribute
+   */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when user changes the input value
+   */
+  onChange: PropTypes.func,
+  /**
+   * Whether the input is required
+   */
+  required: PropTypes.bool,
```

Comment:
https://www.w3schools.com/tags/att_input_required.asp

It should probably stay `required` to match the html attribute